### PR TITLE
feat(css-vars): add function to convert the theme into css variables

### PR DESCRIPTION
**Pull request by lucasaarcoverde**
#### What is the purpose of this pull request?

Adds a function to convert the theme into CSS Variables

#### Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Quality improvement (tests or refactors)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Trivial change (small fix or feature that doesn't impact functionalities)
- [ ] Requires change to documentation, which has been updated accordingly

### CODE REVIEW

File: packages/admin-ui-core/src/createTheme.ts [hunk 1]
```diff
@@ -1,6 +1,11 @@
 import { pick, omit, get } from '@vtex/admin-ui-util'
 
 const constants = {
+  /**
+   * When converting tokens with those namespaces to CSS Variables the algorithm will look
+   * to the color values and replace all usage of color arbitrary values with its correspondent color variable.
```
Comment (Lenderson) [inline packages/admin-ui-core/src/createTheme.ts:6]: I think **by** it is better for this case
```suggestion
   * to the color values and replace all usage of color arbitrary values by its correspondent color variable.
```

File: packages/admin-ui-core/src/createTheme.ts [hunk 2]
```diff
@@ -235,3 +240,86 @@ export function objectToVars(obj: Record<string, any>, parent = '') {
...
+    if (result.includes(colorValue)) {
+      result = result.replaceAll(colorValue, toVarValue(`colors-${key}`))
+    }
+  })
+
+  return result
+}
+
+/**
+ * Parses an theme recursivelly to css variables, joining the paths
```
Comment (Lenderson) [inline packages/admin-ui-core/src/createTheme.ts:284]:
```suggestion
 * Parses a theme recursively to css variables, joining the paths
```

File: packages/admin-ui-core/src/createTheme.ts [hunk 3]
```diff
@@ -235,3 +240,86 @@ export function objectToVars(obj: Record<string, any>, parent = '') {
...
+ *
+ * // returns:
+ * {
+ *   '--admin-ui-colors-blue': 'blue',
+ *   '--admin-ui-colors-yellow': 'yellow',
+ *   '--admin-ui-bg-primary': 'var(--admin-ui-colors-blue)',
+ *   '--admin-ui-bg-secondary': 'var(--admin-ui-colors-yellow)',
+ * }
+ */
+export function generateVars<T>(obj: T, theme = {}, ruleId = '', parent = '') {
```
Comment (Lenderson) [inline packages/admin-ui-core/src/createTheme.ts:305]: **Suggestion**: Maybe changing the `parent` param name could be good, I was expecting this param to be a object, but it is the key name which represents the "parent object" or something like it.

**Review by Lenderson — APPROVED** [review]
LGTM, just a few suggestions.

File: packages/admin-ui-core/src/createTheme.ts (hunk 1 above)
Comment (anitavincent) [reply packages/admin-ui-core/src/createTheme.ts:6]: Replace with is ok too

**Review by anitavincent — APPROVED** [review]

File: packages/admin-ui-core/src/createTheme.ts (hunk 1 above)
Comment (lucasaarcoverde) [reply packages/admin-ui-core/src/createTheme.ts]: I've changed this a little bit and I think `with` fits better, can you take a look?

**Review by Lenderson — APPROVED** [review]

**Pull request merged**